### DISCUSSION TITLE
Possible to scroll widget?

### DISCUSSION THREAD

**Discussion by prurph** [johnlindquist/kit#985] [Q&A]
OS: macOS 12.6
ScriptKit: 1.39.24

When I open a widget I cannot scroll the content in it using either the mouse or keyboard. I've tried wrapping everything in divs with overflow-scroll but to no avail. Is this an intended feature? Thanks!

**Comment by johnlindquist** [comment]
Can you try this build?

https://github.com/johnlindquist/kitapp/releases/tag/v1.40.2

Widgets will now have a "containerClass" property that you can customize with tailwind classes. It will now default to:

 "overflow-auto flex justify-center items-center v-screen h-screen"

https://github.com/johnlindquist/kit/blob/49817381dbd39de51de0ddc5ed36534f9b7151c8/src/api/pro.ts#L21

So it should scroll by default

**Comment by prurph** [comment]
Thanks so much, that works! I would recommend removing `items-center` from the default as for longer content it makes it impossible to scroll high enough to see the top.

Also, I did notice the typedefs are not updated so editors/LSPs will complain that `containerClass` isn't a property of `WidgetOptions`. I tried Debug > Force repair Kit SDK but it persisted. Forgive me if you're already aware of this and just releasing a quick fix. I really appreciate it! 🙌

**Reply by johnlindquist** [reply]
@prurph Thanks for catching the typedefs, they accidentally didn't make it in the commit.

I don't know about removing "items-center", that's been the default since the beginning. I'm not sure if that would impact anyone's current widgets...

**Reply by prurph** [reply]
Ah okay that makes sense--it seems reasonable to leave it and users can override it.

My CSS-fu is close to non-existent but I think in any case where this would affect the ability to scroll all the way to the top of the content, the prior behavior would have already had that content cut off, so it's not as if leaving the justify is a regression.

Thanks again!
